syndeo-template/src/test/java/com/google/cloud/syndeo/it/KafkaToBigQueryIT.java

```diff
@@ -0,0 +1,347 @@
+      dataGeneratorPipeline.cancel();
+      throw new RuntimeException(
+          "Data generator pipeline did not start by deadline. Cancelling test");
+    }
+  }
+
+  @Test
+  public void testKickOffKafkaToBigQuerySyndeoPipeline() throws Exception {
+    // Make sure that Kafka Server exists
+    // Start data generation pipeline
+    PipelineResult dataGeneratorPipeline = kickstartDataGeneration();
+    if (!DataflowUtils.waitUntilState(
+        dataGeneratorPipeline, PipelineResult.State.RUNNING, SIX_MINUTES_MILLIS)) {
+      dataGeneratorPipeline.cancel();
+      throw new RuntimeException(
+          "Data generator pipeline did not start by deadline. Cancelling test");
+    }
+
+    // Start syndeo pipeline.
+    {
+      DataflowTemplateClient.JobInfo syndeoPipeline = kickstartSyndeoPipeline();
+      FlexTemplateClient templateClient =
+          FlexTemplateClient.builder().setCredentials(CREDENTIALS).build();
+      DataflowOperator operator = new DataflowOperator(templateClient);
+
+      operator.waitForCondition(
+          DataflowOperator.Config.builder()
+              .setTimeoutAfter(java.time.Duration.ofMinutes(10))
+              .setJobId(syndeoPipeline.jobId())
+              .setProject(PROJECT)
+              .setRegion(REGION)
+              .build(),
+          () -> {
+            try {
+              DataflowTemplateClient.JobState jobState =
+                  templateClient.getJobStatus(PROJECT, REGION, syndeoPipeline.jobId());
+              return jobState.equals(DataflowTemplateClient.JobState.RUNNING);
+            } catch (IOException e) {
+              // If we were unable to check on the job, then we exit the loop.
+              throw new RuntimeException(e);
+            }
+          });
+      // Wait for two minutes while the pipeline executes
+      Thread.sleep(2 * ONE_MINUTE_MILLIS);
+      templateClient.drainlJob(PROJECT, REGION, syndeoPipeline.jobId());
+      // Wait three minutes while the pipeline drains
+      operator.waitForCondition(
+          DataflowOperator.Config.builder()
+              .setTimeoutAfter(java.time.Duration.ofMinutes(3))
+              .setJobId(syndeoPipeline.jobId())
+              .setProject(PROJECT)
+              .setRegion(REGION)
+              .build(),
+          () -> {
+            try {
+              DataflowTemplateClient.JobState jobState =
+                  templateClient.getJobStatus(PROJECT, REGION, syndeoPipeline.jobId());
+              return DataflowTemplateClient.JobState.DONE_STATES.contains(jobState);
+            } catch (IOException e) {
+              // If we were unable to check on the job, then we exit the loop.
+              throw new RuntimeException(e);
+            }
+          });
+    }
+
+    // Start a new syndeo pipeline. We do this to test for appropriate behavior on restarts
+    DataflowTemplateClient.JobInfo syndeoPipeline = kickstartSyndeoPipeline();
+    FlexTemplateClient templateClient =
```

Comment:
Is there a reason we need different clients?

Reply:
nope. removed the second one.